File: demo/backend/models/gallery_image.go
package models

import "time"

// GalleryImage stores metadata for uploaded images.
type GalleryImage struct {
	ID          uint      `gorm:"primaryKey" json:"id"`
	Key         string    `gorm:"type:text;not null" json:"key"`
	Title       string    `gorm:"type:text;not null" json:"title"`
	ContentType string    `gorm:"type:text;not null" json:"content_type"`
	Size        int       `gorm:"not null" json:"size"`
	UploadedBy  string    `gorm:"type:text;not null" json:"uploaded_by"`
	CreatedAt   time.Time `gorm:"autoCreateTime" json:"created_at"`
}

File: c9dbe538ce0130ca9dd4558253d104edff9b0f73/app/models/gallery_image.py
from datetime import datetime

from sqlalchemy import Integer, Text, DateTime
from sqlalchemy.orm import Mapped, mapped_column
from sqlalchemy.sql import func

from app.database import Base


class GalleryImage(Base):
    __tablename__ = "gallery_images"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    key: Mapped[str] = mapped_column(Text, unique=True, nullable=False)
    title: Mapped[str] = mapped_column(Text, nullable=False)
    content_type: Mapped[str] = mapped_column(Text, nullable=False)
    size: Mapped[int] = mapped_column(Integer, nullable=False)
    uploaded_by: Mapped[str] = mapped_column(Text, nullable=False)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), nullable=False
    )
